sdks/python/apache_beam/io/filebasedsource_test.py

```diff
@@ -256,6 +257,28 @@ def test_estimate_size_of_pattern(self):
     fbs = LineSource(pattern)
     self.assertEquals(17 * 6, fbs.estimate_size())
 
+  def test_estimate_size_with_sampling_same_size(self):
+    num_files = 2 * FileBasedSource.MAX_NUMBER_OF_FILES_FOR_AN_EXACT_STAT
+    pattern, _ = write_pattern([10] * num_files)
+    self.assertEqual(
+        6 * 10 * num_files, FileBasedSource(pattern).estimate_size())
+
+  def test_estimate_size_with_sampling_different_sizes(self):
+    num_files = 2 * FileBasedSource.MAX_NUMBER_OF_FILES_FOR_AN_EXACT_STAT
+    base_size = 100
+    variance = 5
+
+    sizes = []
+    for _ in xrange(num_files):
+      sizes.append(int(random.uniform(base_size - variance,
+                                      base_size + variance)))
+    pattern, _ = write_pattern(sizes)
+    tolerance = 0.2
+    self.assertAlmostEqual(
+        base_size * 6 * num_files,
+        FileBasedSource(pattern).estimate_size(),
+        delta=base_size * 6 * num_files * tolerance)
```

Comment:
That's a pretty big delta. An alternative would be to make this deterministic using random.seed. Doing this you could also test that some files are *not* sampled.